uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -574,15 +627,42 @@ define([
       vm.pagination.pageChanged();
     }
 
+    /**
+     * Performs a set of manipulations to prepare all data for calculation and
+     * then triggers the calculation itself if it can be calculated
+     *
+     * @return {Promise}
+     */
+    function attemptCalculateBalanceChange () {
+      updateOpeningBalance();
+      setHoursDeductionsToRequest();
+      setMinMaxDatesToUI();
+      setDatesToRequest();
```

Comment:
another thing out of place, why are we setting the request dates when we are about to calculate the balance change?  This needs to be moved somewhere else where it makes more sense

Reply:
I am not sure how to solve this to be honest. The problem is that dates are dependent on min/max dates and min/max dates are dependent on absence period.

Reply:
to be discussed

Reply:
Done.